Wikipedia/Code/WMFArticleImageGalleryDataSource.m

```diff
@@ -26,7 +27,28 @@ @implementation WMFArticleImageGalleryDataSource
 @dynamic emptyView;
 
 - (instancetype)initWithArticle:(MWKArticle*)article {
-    self = [super wmf_initWithItemsAndReverseIfNeeded:article.images.uniqueLargestVariants];
+    
+    NSArray* images = [article.images.uniqueLargestVariants bk_select:^BOOL (MWKImage* image) {
+        // Keep image if it's the article image even if we can't determine its size - we
+        // always want to show article image as first "lead" image by gallery.
+        if ([article.image isEqualToImage:image]) {
+            return YES;
+        }
+        
+        if (!image.width || !image.height) {
+            // HAX: if this image MWKImage record doesn't have a width and height value (because
+            // it wasn't determined by parsing it from the article HTML's image url) see if the
+            // cache can tell us the size.
+            UIImage* img = [[WMFImageController wmf_appImageCache] imageFromDiskCacheForKey:image.sourceURLString];
```

Comment:
@bgerstle 

Yes there are such cases. 

Any time an image size is requested which exceeds the width or height canonical image the image tag html won't have a width or height attribute - so we can't catch it at parse time. For instance, this image on "enwiki > color wheel":

<img width="487" alt="screen shot 2016-02-05 at 4 08 39 pm" src="https://cloud.githubusercontent.com/assets/3143487/12862725/0c8f7e00-cc23-11e5-9a78-1a0a6486cd05.png">

Without this code that ^ image disappears from the gallery. 

Thankfully imageFromDiskCacheForKey first tries imageFromMemoryCacheForKey.